processor/spanmetricsprocessor/processor.go

```diff
@@ -136,9 +149,9 @@ func mapDurationsToMillis(vs []time.Duration) []float64 {
 
 // validateDimensions checks duplicates for reserved dimensions and additional dimensions. Considering
 // the usage of Prometheus related exporters, we also validate the dimensions after sanitization.
-func validateDimensions(dimensions []Dimension) error {
+func validateDimensions(dimensions []KeyValuePair, defaults []string) error {
 	labelNames := make(map[string]struct{})
-	for _, key := range []string{serviceNameKey, spanKindKey, statusCodeKey} {
+	for _, key := range defaults {
 		labelNames[key] = struct{}{}
 		labelNames[sanitize(key)] = struct{}{}
 	}
```

Comment:
This section of code could be optimised to increase performance by using indexing instead of a for each style loop.

```suggestion
for i := 0; i < len(defaults); i++ {
    labelNames[defaults[i]] = struct{}{}
    labelNames[sanitize(defaults[i])] = struct{}{}
}
```
The reason for this is that for each style loop creates a new copy on each iteration and we can reduce it here.